src/ui/customBreakpointsUI.ts

```diff
@@ -93,34 +164,92 @@ export function registerCustomBreakpointsUI(
 
-  vscode.window.createTreeView(CustomViews.BrowserBreakpoints, {
+  const view = vscode.window.createTreeView(CustomViews.EventListenerBreakpoints, {
     treeDataProvider: provider,
+    showCollapseAll: true,
+    manageCheckboxStateManually: true,
   });
+
+  view.onDidChangeCheckboxState(e => {
+    if (e.items.length == 1 && e.items[0][0] instanceof BreakpointCategory) {
+      if (e.items[0][1]) {
+        provider.addBreakpoints(e.items[0][0].children);
+      } else {
+        provider.removeBreakpoints(e.items[0][0].children);
+      }
+      return;
+    }
+
+    for (const item of e.items) {
+      if (item[0] instanceof Breakpoint) {
+        if (item[1]) {
+          provider.addBreakpoints([item[0]]);
+        } else {
+          provider.removeBreakpoints([item[0]]);
+        }
+      }
+    }
+  });
+
+  context.subscriptions.push(view);
+
   context.subscriptions.push(
     vscode.commands.registerCommand(Commands.AddCustomBreakpoints, () => {
-      const quickPick = vscode.window.createQuickPick();
-      const items = provider.breakpoints.filter(b => !b.enabled);
-      quickPick.items = items;
-      quickPick.onDidAccept(() => {
-        provider.addBreakpoints(quickPick.selectedItems as Breakpoint[]);
-        quickPick.dispose();
-      });
-      quickPick.show();
+      const items = provider.breakpointCategories
+        .sort(BreakpointCategory.compare)
+        .map(bC => bC.children.map(b => ({ id: b.id, label: `${bC.label}: ${b.label}` })))
+        .flat();
+      vscode.window
+        .showQuickPick(items, {
+          canPickMany: true,
+          placeHolder: 'Select breakpoints to enable',
+        })
+        .then(items => {
+          if (!items) return;
+          const ids = items.map(item => item.id);
+
+          provider.addBreakpoints(
+            provider.breakpointCategories
+              .map(b => b.children)
+              .flat()
+              .filter(b => ids.includes(b.id)),
+          );
+        });
     }),
   );
 
   context.subscriptions.push(
     vscode.commands.registerCommand(Commands.RemoveAllCustomBreakpoints, () => {
-      provider.removeBreakpoints(provider.breakpoints.filter(b => b.enabled).map(b => b.id));
+      provider.removeBreakpoints(
+        provider.breakpointCategories
+          .filter(b => b.checkboxState == vscode.TreeItemCheckboxState.Checked)
+          .map(e => e.children)
+          .flat(),
+      );
     }),
   );
 
   context.subscriptions.push(
-    vscode.commands.registerCommand(
-      Commands.RemoveCustomBreakpoint,
-      (treeItem: vscode.TreeItem) => {
-        provider.removeBreakpoints([treeItem.id as string]);
-      },
-    ),
+    vscode.commands.registerCommand(Commands.RemoveCustomBreakpoints, () => {
+      const items = provider.breakpointCategories
+        .sort(BreakpointCategory.compare)
+        .map(bC => bC.children.map(b => ({ id: b.id, label: `${bC.label}: ${b.label}` })))
+        .flat();
```

Comment:
Can probably have a helper function to make this list, reused in `AddCustomBreakpoints`, with some `filter` function here.